src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Configuration/OpenIdConnectConfigurationValidator.cs

```diff
@@ -0,0 +1,99 @@
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using Microsoft.IdentityModel.Logging;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Microsoft.IdentityModel.Protocols.OpenIdConnect.Configuration
+{
+    /// <summary>
+    /// Defines a class for validating the OpenIdConnectConfiguration by using default policy.
+    /// </summary>
+    public class OpenIdConnectConfigurationValidator : IConfigurationValidator<OpenIdConnectConfiguration>
+    {
+        private int _minimumNumberOfKeys = DefaultMinimumNumberOfKeys;
+
+        /// <summary>
+        /// 1 is the default minimum number of keys.
+        /// </summary>
+        private const int DefaultMinimumNumberOfKeys = 1;
+
+        /// <summary>
+        /// Validates a OpenIdConnectConfiguration by using current policy.
+        /// </summary>
+        /// <param name="openIdConnectConfiguration">The OpenIdConnectConfiguration to validate.</param>
+        /// <returns>A <see cref="ConfigurationValidationResult"/> that contains validation result.</returns>
+        public ConfigurationValidationResult Validate(OpenIdConnectConfiguration openIdConnectConfiguration)
+        {
+            if (openIdConnectConfiguration == null)
+                throw new ArgumentNullException(nameof(openIdConnectConfiguration));
+
+            if (openIdConnectConfiguration.JsonWebKeySet == null || openIdConnectConfiguration.JsonWebKeySet.Keys.Count == 0)
+            {
+                return new ConfigurationValidationResult
+                {
+                    Exception = new ConfigurationValidationException(LogMessages.IDX21817),
+                    Succeeded = false
+                };
+            }
+
+            if (openIdConnectConfiguration.SigningKeys.Count < MinimumNumberOfKeys)
```

Comment:
I think this can be simplified.
If openIdConnectConfiguration.JsonWebKeySet.Count != openIdConnectConfiguration.JsonWebKeySet.Keys.Count, then some of the keys could not be converted from a JsonWebKey to a SecurityKey.
To determine which of those that were unable to be converted, just check the property on the JsonWebKey.ConvertedSecurityKey, which will be set when the the JsonWebKey has been converted.
This happens here: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/adc8d59d4aaae502f77fd8f758e4f020045f69eb/src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Configuration/OpenIdConnectConfigurationRetriever.cs#L100